src/overlays/actors/ovl_En_Cha/z_en_cha.c

```diff
@@ -21,16 +24,98 @@ const ActorInit En_Cha_InitVars = {
     (ActorFunc)EnCha_Update,
     (ActorFunc)EnCha_Draw
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Cha_0x80BEB520/EnCha_Init.asm")
+static ColliderCylinderInit sCylinderInit = {
+    { 
+        COLTYPE_WOOD, 
+        AT_NONE, 
+        AC_ON | AC_HARD | AC_TYPE_PLAYER, 
+        OC1_ON | OC1_TYPE_ALL, 
+        OC2_TYPE_1, 
+        COLSHAPE_CYLINDER, 
+        },
+    { 
+        ELEMTYPE_UNK0, 
+        { 0x00000000, 0x00, 0x00 }, 
+        { 0xF7CFFFFF, 0x00, 0x00 }, 
+        TOUCH_NONE | TOUCH_SFX_NORMAL, 
+        BUMP_ON, 
+        OCELEM_ON, },
+    { 10, 40, 0, { 0, 0, 0 } },
+};
+
+extern UNK_TYPE4 D_06000710;
+extern UNK_TYPE4 D_06000958;
+
+void EnCha_Init(Actor *thisx, GlobalContext *globalCtx) {
+    EnCha *this = THIS;
+    s32 pad;
+
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.colChkInfo.mass = 0xFF;
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.home.rot.z = 0;
+    this->actionFunc = func_80BEB654;
+    this->actor.home.rot.x = this->actor.home.rot.z;
+    gSaveContext.perm.weekEventReg[60] &= 0xFB;
+}
+
+void EnCha_Destroy(Actor *thisx, GlobalContext *globalCtx) {
+    EnCha *this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80BEB5DC(EnCha *this, GlobalContext *globalCtx) {
+    func_80BEB654(this, globalCtx);
+    if (this->actor.cutscene == -1) {
+        this->actionFunc = func_80BEB654;
+        return;
+    }
+    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->actor.cutscene, &this->actor);
+        this->actionFunc = func_80BEB654;
+        return;
+    }
+    ActorCutscene_SetIntentToPlay(this->actor.cutscene);
```

Comment:
```suggestion
    if (this->actor.cutscene == -1) {
        this->actionFunc = func_80BEB654;
    }
    else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
        ActorCutscene_StartAndSetUnkLinkFields(this->actor.cutscene, &this->actor);
        this->actionFunc = func_80BEB654;
    } else {
        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
    }
```
Just out of curiosity, does this work to get rid of the returns?